pandas/tests/indexes/datetimes/test_ops.py

```diff
@@ -460,6 +460,20 @@ def test_add_dti_dti(self):
         with pytest.raises(TypeError):
             dti + dti_tz
 
+    def test_add_dti_ndarray(self):
+        # GH 17558
+        # Check that tz-aware DatetimeIndex + np.array(dtype="timedelta64")
+        # and DatetimeIndex + TimedeltaIndex work as expected
+        dti = pd.DatetimeIndex([pd.Timestamp("2017/01/01")])
```

Comment:
you can move this entire test to ``datetimelike.py``; it will hit datetime, period, timedelta that way

Reply:
we should prob move more of these add tests there as well, but can do that in another PR